indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -0,0 +1,2800 @@
+
+
+  private final Set<PartitionType> subsequentlyDiscoveredPartitions = new HashSet<>();
+  private final TaskStorage taskStorage;
+  private final TaskMaster taskMaster;
+  private final IndexerMetadataStorageCoordinator indexerMetadataStorageCoordinator;
+  private final SeekableStreamIndexTaskClient<PartitionType, SequenceType> taskClient;
+  private final SeekableStreamSupervisorSpec spec;
+  private final String dataSource;
+  private final SeekableStreamSupervisorIOConfig ioConfig;
+  private final SeekableStreamSupervisorTuningConfig tuningConfig;
+  private final SeekableStreamTuningConfig taskTuningConfig;
+  private final String supervisorId;
+  private final TaskInfoProvider taskInfoProvider;
+  private final long futureTimeoutInSeconds; // how long to wait for async operations to complete
+  private final RowIngestionMetersFactory rowIngestionMetersFactory;
+  private final ExecutorService exec;
+  private final ScheduledExecutorService scheduledExec;
+  private final ScheduledExecutorService reportingExec;
+  private final ListeningExecutorService workerExec;
+  private final BlockingQueue<Notice> notices = new LinkedBlockingDeque<>();
+  private final Object stopLock = new Object();
+  private final Object stateChangeLock = new Object();
+  private final Object recordSupplierLock = new Object();
+
+  private final boolean useExclusiveStartingSequence;
+  private boolean listenerRegistered = false;
+  private long lastRunTime;
+  private int initRetryCounter = 0;
+  private volatile DateTime firstRunTime;
+  private volatile RecordSupplier<PartitionType, SequenceType> recordSupplier;
+  private volatile boolean started = false;
+  private volatile boolean stopped = false;
+  private volatile boolean lifecycleStarted = false;
+
+
+  public SeekableStreamSupervisor(
+      final String supervisorId,
+      final TaskStorage taskStorage,
+      final TaskMaster taskMaster,
+      final IndexerMetadataStorageCoordinator indexerMetadataStorageCoordinator,
+      final SeekableStreamIndexTaskClientFactory<? extends SeekableStreamIndexTaskClient<PartitionType, SequenceType>> taskClientFactory,
+      final ObjectMapper mapper,
+      final SeekableStreamSupervisorSpec spec,
+      final RowIngestionMetersFactory rowIngestionMetersFactory,
+      final boolean useExclusiveStartingSequence
+  )
+  {
+    this.taskStorage = taskStorage;
+    this.taskMaster = taskMaster;
+    this.indexerMetadataStorageCoordinator = indexerMetadataStorageCoordinator;
+    this.sortingMapper = mapper.copy().configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true);
+    this.spec = spec;
+    this.rowIngestionMetersFactory = rowIngestionMetersFactory;
+    this.useExclusiveStartingSequence = useExclusiveStartingSequence;
+
+    this.dataSource = spec.getDataSchema().getDataSource();
+    this.ioConfig = spec.getIoConfig();
+    this.tuningConfig = spec.getTuningConfig();
+    this.taskTuningConfig = this.tuningConfig.copyOf();
```

Comment:
This looks very confused. How about adding a new method like `SeekableStreamTuningConfig createTaskTuningConfigFrom(SeekableStreamSupervisorTuningConfig supervisorTuningConfig)` to this class?

Reply:
I've refactored this and added a `convertToTaskTuningConfig()` method to `SeekableStreamSupervisorTuningConfig`